.github/workflows/report.yml

```diff
@@ -1,26 +1,40 @@
 name: 'Test Report'
 on:
   workflow_run:
-    workflows: ['Build App']                     # runs after CI workflow
+    workflows: ['Test and Build'] # runs after CI workflow
+    branches:
+      - dev
+      - main
```

Comment:
Why do you only want to run tests this way? The dependency to the test-and build workflow should be sufficient.